microsoft/testsuites/stress/stress_ng_suite.py

```diff
@@ -0,0 +1,165 @@
+        else:
+            raise SkippedException("No jobfile provided for stress-ng")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'cpu' class stressors",
+        priority=4,
+        timeout=3600,
+    )
+    def verify_stress_ng_cpu_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "cpu")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'memory' class stressors",
+        priority=4,
+        timeout=3600,
+    )
+    def verify_stress_ng_memory_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "memory")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'vm' class stressors",
+        priority=4,
+        timeout=3600,
+    )
+    def verify_stress_ng_vm_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "vm")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'io' class stressors",
+        priority=4,
+        timeout=3600,
+    )
+    def verify_stress_ng_io_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "io")
+
+    @TestCaseMetadata(
+        description="Runs stress-ng's 'network' class stressors",
+        priority=4,
+        timeout=3600,
+    )
+    def verify_stress_ng_network_stressors(
+        self,
+        environment: Environment,
+    ) -> None:
+        self._run_stressor_class(environment, "network")
+
+    def _run_stressor_class(self, environment: Environment, class_name: str) -> None:
+        nodes = [cast(RemoteNode, node) for node in environment.nodes.list()]
+        procs: List[Process] = []
+        for node in nodes:
+            procs.append(node.tools[StressNg].launch_class_async(class_name))
+        for proc in procs:
+            proc.wait_result(timeout=3600, expected_exit_code=0)
+
+    def _run_stress_ng_job(
+        self,
+        job_file: str,
+        environment: Environment,
+        test_result: TestResult,
+    ) -> None:
+        nodes = [cast(RemoteNode, node) for node in environment.nodes.list()]
+        procs: List[Process] = []
+        job_file_name = Path(job_file).name
+        test_status = TestStatus.QUEUED
+        try:
+            for node in nodes:
+                remote_working_dir = node.working_path / "stress_ng_jobs"
+                node.shell.mkdir(remote_working_dir, exist_ok=True)
+                job_file_dest = remote_working_dir / job_file_name
+                node.shell.copy(PurePath(job_file), job_file_dest)
+                procs.append(node.tools[StressNg].launch_job_async(str(job_file_dest)))
+            for proc in procs:
+                proc.wait_result(expected_exit_code=0)
+            test_status = TestStatus.PASSED
+        except Exception:
+            test_status = TestStatus.FAILED
```

Comment:
Add failure reason as possible for failed test cases.

Reply:
Done.